AmplifyPlugins/API/AWSAPICategoryPlugin/AWSAPIPlugin+Reachability.swift

```diff
@@ -0,0 +1,37 @@
+//
+// Copyright 2018-2019 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+import Foundation
+import Combine
+import Reachability
+
+extension AWSAPIPlugin {
+
+    @available(iOS 13.0, *)
+    public func reachability(apiName: String, allowsCellularAccess: Bool) -> AnyPublisher<ReachabilityUpdate, Never>? {
+        guard let baseUrl = self.pluginConfig.endpoints[apiName]?.baseURL,
+            let host = baseUrl.host,
+            var reachabilityMap = self.reachabilityMap else {
+                return nil
+        }
+        if let networkReachability = reachabilityMap[apiName] {
+            return networkReachability.publisher
+        }
+
+        do {
+            let networkReachability = try NetworkReachabilityNotifier(host: host,
+                                                                      allowsCellularAccess: allowsCellularAccess,
+                                                                      reachabilityFactory: Reachability.self)
+            reachabilityMap[apiName] = networkReachability
+            return networkReachability.publisher
+        } catch {
+            return nil
```

Comment:
can we log something here with the local `error` in the catch block? `Amplify.API.log.warn("Failed to create network reachability notifier with \(error))`

Reply:
There's already one in NetworkReachabilityNotifier, but seems to log the error in this class instead... marking as resolved.